pkgs/test_core/lib/src/util/package_config.dart

```diff
@@ -42,3 +43,24 @@ Future<Uri> absoluteUri(String path) async {
 final Future<Package> currentPackage = () async {
   return (await currentPackageConfig).packageOf(await packageConfigUri)!;
 }();
+
+/// Returns the names of all the packages in the workspace.
+///
+/// The returned set includes the package's own name (if the package is not a
+/// workspace, that will be the only name in the set).
+Set<String> workspacePackageNames(Package package) =>
+    _getAllWorkspaceNames(package.root, <String>{});
+
+Set<String> _getAllWorkspaceNames(Uri packageRoot, Set<String> results) {
+  final pubspecUri = packageRoot.resolve('pubspec.yaml');
+  final pubspecFile = File(pubspecUri.toFilePath());
+  if (pubspecFile.existsSync()) {
+    final yaml = pubspecFile.readAsStringSync();
+    final pubspec = Pubspec.parse(yaml, sourceUrl: pubspecUri);
+    results.add(pubspec.name);
+    for (final workspace in pubspec.workspace ?? <String>[]) {
```

Comment:
[nit] According to the docs the `workspace` collection holds  "subpackages", not "workspaces"


```suggestion
    for (final package in pubspec.workspace ?? const <String>[]) {
```